src/main/java/com/pinterest/secor/io/impl/JsonORCFileReaderWriterFactory.java

```diff
@@ -0,0 +1,188 @@
+package com.pinterest.secor.io.impl;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.io.compress.Lz4Codec;
+import org.apache.hadoop.io.compress.SnappyCodec;
+import org.apache.orc.CompressionKind;
+import org.apache.orc.OrcFile;
+import org.apache.orc.Reader;
+import org.apache.orc.RecordReader;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.Writer;
+import org.apache.orc.impl.ZlibCodec;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.pinterest.secor.common.FileRegistry;
+import com.pinterest.secor.common.LogFilePath;
+import com.pinterest.secor.common.SecorConfig;
+import com.pinterest.secor.io.FileReader;
+import com.pinterest.secor.io.FileReaderWriterFactory;
+import com.pinterest.secor.io.FileWriter;
+import com.pinterest.secor.io.KeyValue;
+import com.pinterest.secor.util.ReflectionUtil;
+import com.pinterest.secor.util.orc.JsonFieldFiller;
+import com.pinterest.secor.util.orc.VectorColumnFiller;
+import com.pinterest.secor.util.orc.VectorColumnFiller.JsonConverter;
+import com.pinterest.secor.util.orc.schema.ORCScehmaProvider;
+
+/**
+ * ORC reader/writer implementation
+ * 
+ * @author Ashish (ashu.impetus@gmail.com)
+ *
+ */
+public class JsonORCFileReaderWriterFactory implements FileReaderWriterFactory {
```

Comment:
User can specify this in common configuration file. Here is the property:
secor.file.reader.writer.factory=com.pinterest.secor.io.impl.JsonORCFileReaderWriterFactory